source/darwin/commoncrypto_aes.c

```diff
@@ -360,6 +360,14 @@ struct aws_symmetric_cipher *aws_aes_ctr_256_new_impl(
     return &cc_cipher->cipher_base;
 }
 
+static int s_gcm_decrypt(struct aws_symmetric_cipher *cipher, struct aws_byte_cursor input, struct aws_byte_buf *out) {
+    if (cipher->tag.buffer == NULL) {
+        return aws_raise_error(AWS_ERROR_INVALID_ARGUMENT);
```

Comment:
this covers the case of calling decrypt on gcm without providing a tag. 

before it would fail on all 3 platforms with AWS_ERROR_INVALID_ARGUMENT, but at different points (in finalize on linux and mac, and in decrypt on win). since win really cares about tag during the first decrypt, im just moving the check earlier on mac/linux and keeping the same error message. the rest of the code treats any errors due to bad input (i.e. keys, iv, tag, etc...) as AWS_ERROR_INVALID_ARGUMENT. i think consistency is more important here. we can discuss switching all errors to invalid state